### PR TITLE
rmv distutils

### LINKED ISSUE #228: [Bug]: distutils.version deprecated
### What happened?

I got a warning that `distutils.version` is deprecated, and the version comparison does not work locally.

### Steps to Reproduce

While writing a conversion for https://datadryad.org/stash/dataset/doi:10.5061%2Fdryad.d2547d840 I used the following code:

```python
from datetime import datetime
from neuroconv.datainterfaces import SpikeGLXRecordingInterface

file_path = "/Users/bendichter/Downloads/doi_10.5061_dryad.d2547d840__v9/Pt01_g0/Pt01_g0_imec0/Pt01.imec0.ap.bin"

interface = SpikeGLXRecordingInterface(file_path)

metadata = interface.get_metadata()
metadata["NWBFile"].update(session_start_time=datetime(1900,1,1))
interface.run_conversion("Pt01_out.nwb", metadata=metadata)
```
This returns a warning and an error.

```
/Users/bendichter/dev/neuroconv/src/neuroconv/tools/spikeinterface/spikeinterface.py:1003: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  distutils.version.LooseVersion(pynwb.__version__) >= "1.3.3"
/Users/bendichter/opt/miniconda3/lib/python3.9/site-packages/setuptools/_distutils/version.py:346: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  other = LooseVersion(other)
```

I am also getting a traceback:

```pytb
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In [4], line 1
----> 1 interface.run_conversion("Pt01_out.nwb", metadata=metadata)

File ~/dev/neuroconv/src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py:173, in BaseRecordingExtractorInterface.run_conversion(self, nwbfile_path, nwbfile, metadata, overwrite, stub_test, starting_time, use_times, write_as, write_electrical_series, es_key, compression, compression_opts, iterator_type, iterator_opts)
    170 else:
    171     recording = self.recording_extractor
--> 173 write_recording(
    174     recording=recording,
    175     nwbfile_path=nwbfile_path,
    176     nwbfile=nwbfile,
    177     metadata=metadata,
    178     overwrite=overwrite,
    179     verbose=self.verbose,
    180     starting_time=starting_time,
    181     use_times=use_times,
    182     write_as=write_as,
    183     write_electrical_series=write_electrical_series,
    184     es_key=es_key or self.es_key,
    185     compression=compression,
    186     compression_opts=compression_opts,
    187     iterator_type=iterator_type,
    188     iterator_opts=iterator_opts,
    189 )

File ~/dev/neuroconv/src/neuroconv/tools/spikeinterface/spikeinterface.py:1003, in write_recording(recording, nwbfile_path, nwbfile, metadata, overwrite, verbose, starting_time, use_times, write_as, es_key, write_electrical_series, write_scaled, compression, compression_opts, iterator_type, iterator_opts)
   1000 if nwbfile is not None:
   1001     assert isinstance(nwbfile, pynwb.NWBFile), "'nwbfile' should be of type pynwb.NWBFile"
   1002 assert (
-> 1003     distutils.version.LooseVersion(pynwb.__version__) >= "1.3.3"
   1004 ), "'write_recording' not supported for version < 1.3.3. Run pip install --upgrade pynwb"
   1005 write_as = "raw" if write_as is None else write_as
   1006 compression = "gzip" if compression is None else compression

File ~/opt/miniconda3/lib/python3.9/site-packages/setuptools/_distutils/version.py:90, in Version.__ge__(self, other)
     89 def __ge__(self, other):
---> 90     c = self._cmp(other)
     91     if c is NotImplemented:
     92         return c

File ~/opt/miniconda3/lib/python3.9/site-packages/setuptools/_distutils/version.py:352, in LooseVersion._cmp(self, other)
    350 if self.version == other.version:
    351     return 0
--> 352 if self.version < other.version:
    353     return -1
    354 if self.version > other.version:

TypeError: '<' not supported between instances of 'str' and 'int'
```

It also looks like the `__version__` method of getting the python version isn't working properly.

```python
import pynwb

print(pynwb.__version__)
```
```
'unknown'
```

Using this method, which only works in 3.8+, works fine:

```python
from importlib.metadata import version
print(version("pynwb"))
```

```
2.2.0+24.g4a9ee922.dirty
```

### Operating System

macOS

### Python Executable

Python

### Python Version

3.9

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/neuroconv/blob/main/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/neuroconv/issues)?

### PR DESCRIPTION
**Pull request by bendichter**
## Motivation

fix #228 

## Checklist

- [x] Have you thoroughly read our [Developer Guide](https://neuroconv.readthedocs.io/en/main/developer_guide.html)?
- [x] Have you ensured the PR description clearly describes the problem and solutions?
- [x] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/neuroconv/pulls) for the same change?
- [x] If this PR fixes an issue, is the first line of the PR description `fix #XXX` where `XXX` is the issue number?
- [x] Did you update the [CHANGLOG.md](https://github.com/catalystneuro/neuroconv/blob/main/CHANGELOG.md) file on your branch to describe your changes?

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
LGTM 👍 - see what CI thinks

**Comment by bendichter** [comment]
I'm mostly worried about Python 3.7. It's tough for me to test locally because there are no Python 3.7 binaries for conda osx-arm64 systems. ([source](https://stackoverflow.com/questions/70205633/cannot-install-python-3-7-on-osx-arm64))

**Comment by bendichter** [comment]
@CodyCBakerPhD I can't merge this without your approval

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**
